src/daemon/dlt_daemon_client.c

```diff
@@ -329,11 +329,17 @@ int dlt_daemon_client_send(int sock,DltDaemon *daemon,DltDaemonLocal *daemon_loc
                                                        size2,
                                                        verbose);
 
-			if((sock==DLT_DAEMON_SEND_FORCE) && !sent)
+            if((sock == DLT_DAEMON_SEND_FORCE) && sent != DLT_RETURN_OK)
 			{
 				return DLT_DAEMON_ERROR_SEND_FAILED;
 			}
-		}
+            //this handles some unawaited errors
+            if (sent == DLT_DAEMON_ERROR_UNKNOWN && sock != DLT_DAEMON_SEND_FORCE && daemon->state == DLT_DAEMON_STATE_SEND_DIRECT)
```

Comment:
what un awaited errors you were facing? wanted to know how to test it

Reply:
Also, this is a tiny bit confusing since "unawaited" is not an English word. Does this mean "unexpected"?

Reply:
"unexpected" is definitely better word. Regarding the unexpected behavior: it is related to change in file: dlt_daemon_client_send_all_multiple (l: 122)